src/measure_theory/borel_space.lean

```diff
@@ -875,24 +915,73 @@ variables [normed_group α] [opens_measurable_space α] [measurable_space β]
 lemma measurable_norm : measurable (norm : α → ℝ) :=
 continuous_norm.measurable
 
-lemma measurable.norm {f : β → α} (hf : measurable f) : measurable (λa, norm (f a)) :=
+lemma measurable.norm {f : β → α} (hf : measurable f) : measurable (λ a, norm (f a)) :=
 measurable_norm.comp hf
 
-lemma measurable_nnnorm : measurable (nnnorm : α → nnreal) :=
+lemma measurable_nnnorm : measurable (nnnorm : α → ℝ≥0) :=
 continuous_nnnorm.measurable
 
-lemma measurable.nnnorm {f : β → α} (hf : measurable f) : measurable (λa, nnnorm (f a)) :=
+lemma measurable.nnnorm {f : β → α} (hf : measurable f) : measurable (λ a, nnnorm (f a)) :=
 measurable_nnnorm.comp hf
 
 lemma measurable_ennnorm : measurable (λ x : α, (nnnorm x : ennreal)) :=
 measurable_nnnorm.ennreal_coe
 
 lemma measurable.ennnorm {f : β → α} (hf : measurable f) :
-  measurable (λa, (nnnorm (f a) : ennreal)) :=
+  measurable (λ a, (nnnorm (f a) : ennreal)) :=
 hf.nnnorm.ennreal_coe
 
 end normed_group
 
+section limits
+
+variables [measurable_space β] [metric_space β] [borel_space β]
+
+open metric
+
+/-- A limit (over a general filter) of measurable `ℝ≥0` valued functions is measurable. -/
+lemma measurable_of_tendsto_nnreal' {ι ι'} {f : ι → α → ℝ≥0} {g : α → ℝ≥0} (u : filter ι)
+  [ne_bot u] (hf : ∀ i, measurable (f i)) (lim : tendsto f u (𝓝 g)) {p : ι' → Prop}
+  {s : ι' → set ι} (hu : u.has_countable_basis p s) (hs : ∀ i, (s i).countable) : measurable g :=
```

Comment:
I wonder if we actually need `hs` in this lemma. If `u` has a countable basis, then we can choose a monotonically decreasing sequence of sets `s i`, `i : ℕ`, as a basis. Choosing `x : ℕ → ι`, `x i ∈ s i`, we get a sequence `f ∘ x` which converges to `g`.

Without `hs` this will work, e.g., for `at_top` on `real`s.

Reply:
You might be right. I saw that this condition came up when giving the natural argument, and held in the only case I wanted to apply it to `at_top` on `nat`.
I'm happy to generalize the statement by dropping that assumption, but I'm not too familiar with the filter library (or the math behind them). Are there theorems in mathlib that are the steps of your argument?

Do you think the same assumption can be dropped in `measurable_liminf'`?

Reply:
It seems that this will add complexity to the proof with no real gain. I think, a comment like "the assumption `hs` can be dropped using `is_countably_generated.has_antimono_basis`" would do.

The assumption can't be dropped in `measurable_liminf'` because `liminf (principal s)` is just `infi`, and `infi` over an uncountable set can be non-measurable.